spec/forms/provider_interface/offer_wizard_spec.rb

```diff
@@ -78,31 +81,35 @@
 
     context 'when changing an offer' do
       let(:decision) { :change_offer }
+      let(:query_service) { instance_double(GetChangeOfferOptions) }
+      let(:provider_user) { instance_double(ProviderUser) }
+      let(:provider_id) { create(:provider).id }
+      let(:course_id) { create(:course).id }
+      let(:course_option_id) { create(:course_option).id }
+
+      before do
+        allow(ProviderUser).to receive(:find).and_return(provider_user)
+        allow(provider_user).to receive(:id).and_return(1)
+        allow(GetChangeOfferOptions).to receive(:new).and_return(query_service)
+        allow(store).to receive(:write)
+      end
 
       context 'when current_step is :select_option' do
         let(:current_step) { :select_option }
 
         context 'when there are multiple available providers' do
-          before do
-            provider_user = instance_double(ProviderUser, providers: build_stubbed_list(:provider, 2))
-            allow(ProviderUser).to receive(:find).and_return(provider_user)
-          end
-
           it 'returns :providers' do
+            allow(query_service).to receive(:available_providers).and_return([create(:provider), create(:provider)])
```

Comment:
```suggestion
            allow(query_service).to receive(:available_providers).and_return(create_list(:provider,2 ))
```

Reply:
We did look into this - we ended up feeling like create_list made it a bit harder to read, as unless you're familiar with the method, you can't immediately see that it results in an array.